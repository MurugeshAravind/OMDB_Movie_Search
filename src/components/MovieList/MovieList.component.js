import React from "react";
import { useState } from "react"
import Navigation from "../../common/Navigation/Navigation"
import Search from "../../common/Search/Search";
import Card from "../../common/Card/Card";
import Loader from "../../common/Loader/Loader";
const MovieList = () => {
    const [searchResults, setSearchResults] = useState([]);
    const [loaderStatus, setLoaderStatus] = useState(false);
    const getSearchResults = (results) => {
        setSearchResults(results)
    }
    const getLoaderStatus = (status) => {
        setLoaderStatus(status)
    }
    return <>
    <Navigation />
    <Search searchResult={getSearchResults} loaderStatus={getLoaderStatus}/>
    {loaderStatus ? <Loader /> : null}
    {(!loaderStatus && searchResults) ? <Card poster={searchResults} /> : null}
    </>
}
export default MovieList