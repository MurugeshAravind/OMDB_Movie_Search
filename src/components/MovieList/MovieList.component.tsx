import React from "react";
import { useState, useRef, useCallback } from "react";
import Navigation from "../../common/Navigation/Navigation";
import Search from "../../common/Search/Search";
import Card from "../../common/Card/Card";
import Loader from "../../common/Loader/Loader";
import useMovieSearch from "../../common/useMovieSearch";
const MovieList = () => {
  const [query, setQuery] = useState<string>("");
  const [pageNumber, setPageNumber] = useState<number>(1);
  const { movies, hasMore, loading, error } = useMovieSearch(query, pageNumber);
  const observer = useRef<any>();
  const lastBookElementRef = useCallback(
    (node: any) => {
      if (loading) return;
      if (observer.current) observer.current.disconnect();
      observer.current = new IntersectionObserver((entries) => {
        if (entries[0].isIntersecting && hasMore && !error) {
          setPageNumber((previousPageNumber) => previousPageNumber + 1);
        }
      });
      if (node) observer.current.observe(node);
    },
    [loading, hasMore, error]
  );

  const getQueryValue = (value: string) => {
    setQuery(value);
  };
  const getPageNumber = (value: number) => {
    setPageNumber(value);
  };
  return (
    <>
      <Navigation />
      <Search queryValue={getQueryValue} page={getPageNumber} />
      {loading ? <Loader /> : null}
      {movies.length > 0 ? (
        <Card letRef={lastBookElementRef} poster={movies} />
      ) : null}
    </>
  );
};
export default MovieList;
