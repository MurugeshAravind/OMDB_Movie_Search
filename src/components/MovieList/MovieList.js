import { useState } from "react"
import Navigation from "../Navigation/Navigation"
import Search from "../Search/Search";
import Card from "../Card/Card";
import Loader from "../Loader/Loader";
const MovieList = () => {
    const [searchResults, setSearchResults] = useState([]);
    const [loaderStatus, setLoaderStatus] = useState(false);
    const getSearchResults = (results) => {
        setSearchResults(results)
    }
    const getLoaderStatus = (status) => {
        setLoaderStatus(status)
    }
    return <>
    <Navigation />
    <Search searchResult={getSearchResults} loaderStatus={getLoaderStatus}/>
    {loaderStatus ? <Loader /> : null}
    {!loaderStatus && searchResults && <Card poster={searchResults} />}
    </>
}
export default MovieList