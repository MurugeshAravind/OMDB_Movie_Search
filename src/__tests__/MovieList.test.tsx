import 'jest-styled-components';
import MovieList from '../components/MovieList/MovieList.component';
import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
const MovieListComponent = render(
  <MemoryRouter future={{ v7_startTransition: true }}>
    <MovieList />
  </MemoryRouter>,
);
describe('movie list test suite', () => {
  test('movie list snapshot test', () => {
    expect(MovieListComponent).toMatchSnapshot();
  });
});
