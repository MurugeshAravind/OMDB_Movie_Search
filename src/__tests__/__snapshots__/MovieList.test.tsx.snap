// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`movie list test suite movie list snapshot test 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <nav
        class="dZuuai"
      >
        <h1
          class="bHFXsR"
        >
          OMDB Search
        </h1>
      </nav>
      <div
        class="bpjZGj"
      >
        <input
          class="gdZrxl"
          data-testid="search-input"
          placeholder="Enter your movie name"
          type="search"
          value=""
        />
      </div>
      <div
        class="btoief"
      />
    </div>
  </body>,
  "container": <div>
    <nav
      class="dZuuai"
    >
      <h1
        class="bHFXsR"
      >
        OMDB Search
      </h1>
    </nav>
    <div
      class="bpjZGj"
    >
      <input
        class="gdZrxl"
        data-testid="search-input"
        placeholder="Enter your movie name"
        type="search"
        value=""
      />
    </div>
    <div
      class="btoief"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
